tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/OverlayAccess.java

```diff
@@ -0,0 +1,327 @@
+		if (button == null) {
+			return;
+		}
+		button.setSelection(true);
+		if (option == SearchOptions.GLOBAL) {
+			button.setSelection(false);
+		}
+		button.notifyListeners(SWT.Selection, null);
+	}
+
+	@Override
+	public void unselect(SearchOptions option) {
+		ToolItem button= getButtonForSearchOption(option);
+		if (button == null) {
+			return;
+		}
+		button.setSelection(false);
+		if (option == SearchOptions.GLOBAL) {
+			button.setSelection(true);
+		}
+		button.notifyListeners(SWT.Selection, null);
+	}
+
+	@Override
+	public void simulateEnterInFindInputField(boolean shiftPressed) {
+		simulateKeyPressInFindInputField(SWT.CR, shiftPressed);
+	}
+
+	@Override
+	public void simulateKeyPressInFindInputField(int keyCode, boolean shiftPressed) {
+		final Event event= new Event();
+		event.type= SWT.KeyDown;
+		event.keyCode= keyCode;
+		if (shiftPressed) {
+			event.stateMask= SWT.SHIFT;
+		}
+		find.notifyListeners(SWT.KeyDown, event);
+		find.traverse(SWT.TRAVERSE_RETURN, event);
+		FindReplaceTestUtil.runEventQueue();
+	}
+
+	@Override
+	public String getFindText() {
+		return find.getText();
+	}
+
+	@Override
+	public String getReplaceText() {
+		return replace.getText();
+	}
+
+	@Override
+	public void setFindText(String text) {
+		find.setText(text);
+		find.notifyListeners(SWT.Modify, null);
+	}
+
+	@Override
+	public void setReplaceText(String text) {
+		openReplaceDialog();
+		replace.setText(text);
+	}
+
+	@Override
+	public ToolItem getButtonForSearchOption(SearchOptions option) {
+		switch (option) {
+			case CASE_SENSITIVE:
+				return caseSensitive;
+			case REGEX:
+				return regEx;
+			case WHOLE_WORD:
+				return wholeWord;
+			case GLOBAL:
+				return inSelection;
+			//$CASES-OMITTED$
+			default:
+				return null;
+		}
+	}
+
+	private Set<SearchOptions> getEnabledOptions() {
+		HashSet<SearchOptions> ret= new HashSet<>();
+
+		for (SearchOptions option : SearchOptions.values()) {
+			ToolItem optionButton= getButtonForSearchOption(option);
+			if (option == SearchOptions.GLOBAL && !optionButton.isEnabled()) {
+				ret.add(option); // The "Global" option is mapped to a button that
+									// selects whether to search in the selection, thus inverting the semantic
+			} else if (optionButton != null && optionButton.isEnabled()) {
+				ret.add(option);
+			}
+		}
+
+		return ret;
+	}
+
+	private Set<SearchOptions> getSelectedOptions() {
+		HashSet<SearchOptions> ret= new HashSet<>();
+
+		for (SearchOptions option : SearchOptions.values()) {
+			ToolItem optionButton= getButtonForSearchOption(option);
+			if (option == SearchOptions.GLOBAL && !optionButton.getSelection()) {
+				ret.add(option); // The "Global" option is mapped to a button that
+									// selects whether to search in the selection, thus inverting the semantic
+			} else if (option != SearchOptions.GLOBAL && optionButton != null && optionButton.getSelection()) {
+				ret.add(option);
+			}
+		}
+
+		return ret;
```

Comment:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/67ef99dd0640b16d57511fe574efb5974ee221e9

The functionality is not as clean here because we need to handle some edge cases in the filter.

Reply:
I have implemented these changes and I agree.
Interestingly, changing 2) has no effect on the validity of the unit tests.